Archives/ArchivePacker.cpp

```diff
@@ -1,28 +1,38 @@
 #include "ArchivePacker.h"
 #include "../XFile.h"
 #include "../StringHelper.h"
-#include <windows.h>
+#include <cstdio>
 #include <stdexcept>
 
 namespace Archives
 {
 	ArchivePacker::ArchivePacker() { }
 	ArchivePacker::~ArchivePacker() { }
 
-	// newFile has it's name changed to that of fileToReplace and the old file is deleted.
-	// Returns true if successful and false otherwise
-	bool ArchivePacker::ReplaceFileWithFile(const char *fileToReplace, const char *newFile)
+	// newFile has its name changed to fileToReplace and the old file is deleted.
+	void ArchivePacker::ReplaceFileWithFile(const std::string& fileToReplace, const std::string& newFile)
 	{
-		std::string tempFileName("Temporary.vol");
-		MoveFileA(fileToReplace, tempFileName.c_str());
+		// If C++17 filesystem comes out of experimental status on both MSVC and Linux compilers, 
+		// consider switching to std::filesystem::rename & std::filesystem::remove
```

Comment:
This makes sense, I didn't realize there was a non-experimental/filesystem version of rename and remove.

Side note, I was initially confused by the identically named `std::remove` (and `std::remove_if`) from `<algorithm>`, though they have different signatures so there is no conflict.

Perhaps we should have wrapper functions for rename and remove in XFile, so as to minimize changes when we switch over. Or rather, since XFile already imports filesystem, we can just have it call the new methods right now.

In particular, the new [`std::experimental::rename`](http://en.cppreference.com/w/cpp/experimental/fs/rename) guarantees semantics such that we can avoid the intermediate temporary filename. That would simplify error checking considerably.

As for compatibility, XFile already imports filesystem, which the MinGW cross compiler fails on. Hence I haven't been testing or reporting issues on that one file. Just now though, I've installed a newer version of Clang, which can handle `#include <experimental/filesystem>`. Naturally though, it fails on `#include <windows.h>`, so again I have yet to find a way to build the entire project from Linux.

I would recommend creating wrappers in XFile, and calling the new experimental/filesystem code from there. We might only need `rename` for this feature. This can be done in a future branch.

Edit: I'm just realizing the scope here. The `rename` method from the new filesystem header can completely replace the current function `ReplaceFileWithFile`. We should dump this current function, and replace calls with an XFile wrapper that makes use of the new `filesystem::rename`.